qiskit/transpiler/preset_passmanagers/builtin_plugins.py

```diff
@@ -0,0 +1,288 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Built-in transpiler stage plugins for preset pass managers."""
+
+from qiskit.transpiler.passmanager_config import PassManagerConfig
+from qiskit.transpiler.passmanager import PassManager
+from qiskit.transpiler.exceptions import TranspilerError
+from qiskit.transpiler.passes import BasicSwap
+from qiskit.transpiler.passes import LookaheadSwap
+from qiskit.transpiler.passes import StochasticSwap
+from qiskit.transpiler.passes import SabreSwap
+from qiskit.transpiler.passes import Error
+from qiskit.transpiler.preset_passmanagers import common
+from qiskit.transpiler.preset_passmanagers.plugin import PassManagerStagePlugin
+
+
+def get_vf2_limit(pass_manager_config):
+    """Get the vf2 call limit for vf2 based layout passes."""
+    vf2_call_limit = None
+    opt_level = pass_manager_config.optimization_level
+    if pass_manager_config.layout_method is None and pass_manager_config.initial_layout is None:
+        if opt_level == 1:
+            vf2_call_limit = int(5e4)  # Set call limit to ~100ms with retworkx 0.10.2
+        elif opt_level == 2:
+            vf2_call_limit = int(5e6)  # Set call limit to ~10 sec with retworkx 0.10.2
+        elif opt_level == 3:
+            vf2_call_limit = int(3e7)  # Set call limit to ~60 sec with retworkx 0.10.2
+    return vf2_call_limit
+
+
+class BasicSwapPassManager(PassManagerStagePlugin):
+    """Plugin class for routing stage with :class:`~.BasicSwap`"""
+
+    def pass_manager(self, pass_manager_config: PassManagerConfig) -> PassManager:
+        """Build routing stage PassManager."""
+        opt_level = pass_manager_config.optimization_level
+        seed_transpiler = pass_manager_config.seed_transpiler
+        target = pass_manager_config.target
+        coupling_map = pass_manager_config.coupling_map
+        backend_properties = pass_manager_config.backend_properties
+        routing_pass = BasicSwap(coupling_map)
+        vf2_call_limit = get_vf2_limit(pass_manager_config)
+        if opt_level == 0:
+            return common.generate_routing_passmanager(
```

Comment:
why not parametrize `generate_routing_passmanager` with `pass_manager_config`? This will avoid this switch and allows `generate_routing_passmanager` to grow in the future. Besides, it is intuitive that a passmanager generator has access to the full `pass_manager_config`.